sound/soc/sof/Makefile

```diff
@@ -4,6 +4,7 @@ ccflags-y += -DDEBUG
 
 snd-sof-objs := core.o ops.o loader.o ipc.o pcm.o pm.o debug.o topology.o\
 		control.o trace.o utils.o sof-audio.o
+snd-sof-$(CONFIG_SND_SOC_SOF_CLIENT) += sof-client.o
```

Comment:
so here if SOF_CLIENT is a boolean then this cannot be used as a module.
that's not right I am afraid.

You would need something similar to HDAudio, where you have a boolean that defines SNd_SOC_SOF_CLIENT_SUPPORT. Then in CONFIG_SOF you select SOC_SOF_CLIENT if SOC_SOF_CLIENT_SUPPORT. that will propagate the m or y from the top level.

Reply:
Thanks for the tutorial @plbossart. I think I understand kconfig a bit better now but I might still fail the test on this subject :)